File: go/count_number_of_homogenous_substrings.go
//no. 1759
import "string"

func countHomogenous(s string) int {
    var ans = 1
    var curr_streak = 1
    const mod = 1e9 + 7
    for i := 0; i < len(s) - 1; i++ {
        if s[i] == s[i + 1]{
            curr_streak += 1
        }else{
            curr_streak = 1
        }
        ans += curr_streak
        ans = ans % mod
    }
    return ans
}

File: C++/count_number_of_homogenous_substrings.cpp
// no. 1759
#include <string>
using namespace std;

class Solution {
public:
    int countHomogenous(string s) {
        int ans = 1;
        int curr_streak = 1;
        int mod = 1e9 + 7;
        for(int i = 0; i < s.length() - 1; i++){
            if(s[i] == s[i + 1]){
                curr_streak ++;
            }else{
                curr_streak = 1;
            }
            ans += curr_streak;
            ans = ans % mod;
        }
        return ans;
    }
};
